src/logic/currencyValues/store/model/currencyValues.ts

```diff
@@ -1,39 +1,46 @@
 import { Record } from 'immutable'
 
-export const AVAILABLE_CURRENCIES = {
-  USD: 'USD',
-  EUR: 'EUR',
-  CAD: 'CAD',
-  HKD: 'HKD',
-  ISK: 'ISK',
-  PHP: 'PHP',
-  DKK: 'DKK',
-  HUF: 'HUF',
-  CZK: 'CZK',
-  AUD: 'AUD',
-  RON: 'RON',
-  SEK: 'SEK',
-  IDR: 'IDR',
-  INR: 'INR',
-  BRL: 'BRL',
-  RUB: 'RUB',
-  HRK: 'HRK',
-  JPY: 'JPY',
-  THB: 'THB',
-  CHF: 'CHF',
-  SGD: 'SGD',
-  PLN: 'PLN',
-  BGN: 'BGN',
-  TRY: 'TRY',
-  CNY: 'CNY',
-  NOK: 'NOK',
-  NZD: 'NZD',
-  ZAR: 'ZAR',
-  MXN: 'MXN',
-  ILS: 'ILS',
-  GBP: 'GBP',
-  KRW: 'KRW',
-  MYR: 'MYR',
+export enum AVAILABLE_CURRENCIES {
+  USD = 'USD',
+  EUR = 'EUR',
+  CAD = 'CAD',
+  HKD = 'HKD',
+  ISK = 'ISK',
+  PHP = 'PHP',
+  DKK = 'DKK',
+  HUF = 'HUF',
+  CZK = 'CZK',
+  AUD = 'AUD',
+  RON = 'RON',
+  SEK = 'SEK',
+  IDR = 'IDR',
+  INR = 'INR',
+  BRL = 'BRL',
+  RUB = 'RUB',
+  HRK = 'HRK',
+  JPY = 'JPY',
+  THB = 'THB',
+  CHF = 'CHF',
+  SGD = 'SGD',
+  PLN = 'PLN',
+  BGN = 'BGN',
+  TRY = 'TRY',
+  CNY = 'CNY',
+  NOK = 'NOK',
+  NZD = 'NZD',
+  ZAR = 'ZAR',
+  MXN = 'MXN',
+  ILS = 'ILS',
+  GBP = 'GBP',
+  KRW = 'KRW',
+  MYR = 'MYR',
+}
+
+export type Currency = keyof typeof AVAILABLE_CURRENCIES
```

Comment:
can't you use plain enum as a type? 
https://www.tutorialsteacher.com/typescript/typescript-enum#:~:text=Enums%20or%20enumerations%20are%20a,be%20numeric%20or%20string%20values.

Reply:
done